.github/workflows/compile.yml

```diff
@@ -0,0 +1,94 @@
+name: compile
+
+on:
+  pull_request_target
```

Comment:
We need to build the code on `pull_request`, but send a notification using another workflow that's run on `pull_request_target`. See this article: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/